source/clear-linux/guides/maintenance/swupd-guide.rst

```diff
@@ -1,185 +1,364 @@
+so there's no ambiguity between two systems running the same version of |CL|.
+
+Updating
+========
+
+|CL| enforces regular updating of the OS by default and will automatically
+check for updates against a version server. The content server provides the
+file and metadata content for all versions and can be the same as the
+version server. The content url server provides metadata in the form of
+manifests. These Manifest files list and describe file contents, symlinks,
+and directories. Additionally, the actual content is
+provided to clients in the form of archive files.
+
+Software updates with |CL| are also efficient. Unlike package-based
+distributions, :command:`swupd` only updates files that have changed rather
+than entire packages. For example, it is quite common for an OS security
+patch to be as small as 15 KB. Using binary deltas, the |CL| is able to
+apply only what is needed.
+
+To get a more detailed understanding of how to generate update content for
+|CL| see the :ref:`mixer <mixer>` tool. 
+
+How it works
+************
+
+Prerequisites
+=============
+
+* The device is on a well-connected network.
+* The device is able to connect to an update server. The default server is:
+  http://update.clearlinux.org
+
+Updates
+=======
+
+|CL| updates are automatic by default but can be set to occur only on
+demand. :command:`swupd` makes sure that regular updates are simple and
+secure. It can also check the validity of currently installed files and
+software and correct any problems.
+
+Manifests
+---------
+
+The Clear Linux OS software update content consists of data and
+metadata.  The data is the files that end up in the OS. The metadata
+contains relevant information to properly provision the data to the OS
+file system, as well as update the system and add or remove additional
+content to the OS.
+
+The Manifests are mostly long lists of hashes that describe content.
+Each bundle gets its own manifest file. There is a master manifest
+file that describes all manifests to tie it all together.
+
+Fullfiles, packs, and delta packs
+---------------------------------
+
+The data that an update provisions to a system can be obtained in
+three different ways. There are three different methods, and they
+exist to optimize the delivery of content and speed up updates.
+
+Fullfiles are always generated for every file in every release. This
+allows any Clear Linux OS to obtain the exact copy of the content
+for each version directly. This would be used if the OS verification
+needed to replace a single file, for instance.
+
+Packs are available for some releases and combine many files to speed
+up the creation of installation media and large updates. Delta packs
+are an optimized version of packs that only contain updates (binary
+diffs) and cannot be used without having the original file content.
+
+Bundle Search
+=============
+
+:command:`swupd` search downloads manifest data and searches for

```

Comment:
swupd seach has changed a bit. And it will change more in next swupd release. Currently search is not searching for matching paths. It looks on different kinds of metadata to determine what's the best bundle to install based on a search term.